packages/client/ui/react-ui/src/providers/auth/AuthFormProvider.test.tsx

```diff
@@ -102,4 +142,24 @@ describe("AuthFormProvider", () => {
         expect(consoleSpy).toHaveBeenCalled();
         consoleSpy.mockRestore();
     });
+
+    test("throws error when web3 login method is used with createOnLogin=all-users", () => {
+        const invalidState = {
+            ...mockInitialState,
+            loginMethods: ["web3"] as LoginMethod[],
+            embeddedWallets: {
+                createOnLogin: "all-users",
+                defaultChain: "base-sepolia",
+                type: "evm-smart-wallet",
+            } as CrossmintAuthWalletConfig,
+        };
+
+        expect(() =>
+            render(
+                <AuthFormProvider initialState={invalidState}>
+                    <TestComponent />
+                </AuthFormProvider>
+            )
+        ).toThrowError("Creating wallets on login is not yet supported for web3 login method");
```

Comment:
so what happens with web3 login? when is the wallet created?

Reply:
we just use the one time code same as using otp / google auth

Reply:
but that's for auth, after authenticating via otp or google auth it will getOrCreateWallet, why not with web3 login?

Reply:
it will with getOrCreateWallet 
just not when `createOnLogin: "all-users"` is set

Reply:
was more of a business decision i _think_